sched/irq/irq_csection.c

```diff
@@ -192,208 +192,207 @@ irqstate_t enter_critical_section(void)
    * lists are valid.
    */
 
-  if (nxsched_get_initstate() >= OSINIT_TASKLISTS)
+  DEBUGASSERT(g_nx_initstate >= OSINIT_TASKLISTS);
```

Comment:
It is better to move DEBUG check into `up_interrupt_context()` block, which keeps `enter/leave_critical_section()` still callable during the initialization phase to avoid unnecessary assertions